sleeping in TestSystemMonitor.test_login_logout_messages is not a solution

```
self = <tests.golem.monitor.test_monitor.TestSystemMonitor testMethod=test_login_logout_messages>

    def test_login_logout_messages(self):
        """Test whether correct login and logout messages
                and protocol data were sent."""
    
        def check(f, msg_type):
            with mock.patch(
                'golem.monitor.transport.httptransport.'
                + 'DefaultHttpSender.post_json') \
                    as mock_send:
                f()
                time.sleep(0.005)
                self.assertEqual(mock_send.call_count, 1)
                result = json.loads(mock_send.call_args[0][0])
                expected_d = {
                    'proto_ver': MONITOR_CONFIG['PROTO_VERSION'],
                    'data': {
                        'type': msg_type,
                        'protocol_versions': {
                            'monitor': self.monitor.config['PROTO_VERSION'],
                            'p2p': variables.PROTOCOL_CONST.ID,
                            'task': variables.PROTOCOL_CONST.ID,
                        },
                        'metadata': {
                            'type': 'NodeMetadata',
                            'net': 'testnet',
                            'timestamp': mock.ANY,
                            'cliid': 'cliid',
                            'sessid': 'sessid',
                            'os': 'os',
                            'version': 'ver',
                            'settings': mock.ANY,
                        },
                        'cliid': 'cliid',
                        'sessid': 'sessid',
                        'timestamp': mock.ANY,
                    }
                }
                self.assertEqual(expected_d, result)
    
>       check(self.monitor.on_login, "Login")

tests/golem/monitor/test_monitor.py:112: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/golem/monitor/test_monitor.py:84: in check
    self.assertEqual(mock_send.call_count, 1)
E   AssertionError: 0 != 1
```
While it works on a developer's PCs, it sometimes fails on external CI.

--- a/tests/golem/monitor/test_monitor.py
+++ b/tests/golem/monitor/test_monitor.py
@@ -12,12 +12,25 @@ from golem import testutils
 from golem.clientconfigdescriptor import ClientConfigDescriptor
 from golem.core import variables
 from golem.monitor.model.nodemetadatamodel import NodeMetadataModel
-from golem.monitor.monitor import SystemMonitor, SenderThread
+from golem.monitor.monitor import SystemMonitor, SenderThread, Sender
 from golem.monitorconfig import MONITOR_CONFIG
 from golem.task.taskrequestorstats import CurrentStats, FinishedTasksStats, \
     EMPTY_FINISHED_SUMMARY
 
 random = Random(__name__)
+
+
+class MockSenderThread:
+    def __init__(self):
+        with mock.patch('golem.monitor.transport.sender.DefaultHttpSender'):
+            self.sender = Sender('some_host', 'some_timeout',
+                                 MONITOR_CONFIG['PROTO_VERSION'])
+
+    def send(self, o):
+        self.sender.send(o)
+
+    def join(self):
+        pass
 
 
 class TestSystemMonitor(TestCase, testutils.PEP8MixIn):
@@ -33,15 +46,12 @@ class TestSystemMonitor(TestCase, testutils.PEP8MixIn):
         meta_data = NodeMetadataModel(
             client_mock, 'os', 'ver')
         config = MONITOR_CONFIG.copy()
-        config['HOST'] = 'http://localhost/88881'
-        config['PING_ME_HOSTS'] = [config['HOST']]
-        config['SENDER_THREAD_TIMEOUT'] = 0.05
+        config['PING_ME_HOSTS'] = ['']
         self.monitor = SystemMonitor(meta_data, config)
-        self.monitor.start()
+        self.monitor._sender_thread = MockSenderThread()
 
     def tearDown(self):
         self.monitor.shut_down()
-        del self.monitor
 
     def test_monitor_messages(self):
         """Just check if all signal handlers run without errors"""
@@ -61,8 +71,8 @@ class TestSystemMonitor(TestCase, testutils.PEP8MixIn):
         ccd = ClientConfigDescriptor()
         ccd.node_name = "new node name"
         client_mock = mock.MagicMock()
-        client_mock.cliid = 'CLIID'
-        client_mock.sessid = 'SESSID'
+        client_mock.get_key_id = mock.MagicMock(return_value='cliid')
+        client_mock.session_id = 'sessid'
         client_mock.config_desc = ccd
         new_meta_data = NodeMetadataModel(
             client_mock, "win32", "1.3")
@@ -74,41 +84,38 @@ class TestSystemMonitor(TestCase, testutils.PEP8MixIn):
             and protocol data were sent."""
 
         def check(f, msg_type):
-            with mock.patch(
-                'golem.monitor.transport.httptransport.'
-                + 'DefaultHttpSender.post_json') \
-                    as mock_send:
-                f()
-                time.sleep(0.005)
-                self.assertEqual(mock_send.call_count, 1)
-                result = json.loads(mock_send.call_args[0][0])
-                expected_d = {
-                    'proto_ver': MONITOR_CONFIG['PROTO_VERSION'],
-                    'data': {
-                        'type': msg_type,
-                        'protocol_versions': {
-                            'monitor': self.monitor.config['PROTO_VERSION'],
-                            'p2p': variables.PROTOCOL_CONST.ID,
-                            'task': variables.PROTOCOL_CONST.ID,
-                        },
-                        'metadata': {
-                            'type': 'NodeMetadata',
-                            'net': 'testnet',
-                            'timestamp': mock.ANY,
-                            'cliid': 'cliid',
-                            'sessid': 'sessid',
-                            'os': 'os',
-                            'version': 'ver',
-                            'settings': mock.ANY,
-                        },
+            f()
+            mock_send = self.monitor._sender_thread.sender.transport.post_json
+            self.assertEqual(mock_send.call_count, 1)
+            result = json.loads(mock_send.call_args[0][0])
+            expected_d = {
+                'proto_ver': MONITOR_CONFIG['PROTO_VERSION'],
+                'data': {
+                    'type': msg_type,
+                    'protocol_versions': {
+                        'monitor': self.monitor.config['PROTO_VERSION'],
+                        'p2p': variables.PROTOCOL_CONST.ID,
+                        'task': variables.PROTOCOL_CONST.ID,
+                    },
+                    'metadata': {
+                        'type': 'NodeMetadata',
+                        'net': 'testnet',
+                        'timestamp': mock.ANY,
                         'cliid': 'cliid',
                         'sessid': 'sessid',
-                        'timestamp': mock.ANY,
-                    }
+                        'os': 'os',
+                        'version': 'ver',
+                        'settings': mock.ANY,
+                    },
+                    'cliid': 'cliid',
+                    'sessid': 'sessid',
+                    'timestamp': mock.ANY,
                 }
-                self.assertEqual(expected_d, result)
+            }
+            self.assertEqual(expected_d, result)
 
         check(self.monitor.on_login, "Login")
+        self.monitor._sender_thread.sender.transport.reset_mock()
         check(self.monitor.on_logout, "Logout")
 
     @mock.patch('requests.post')

